### ISSUE TITLE
Predict depends on batch size

### ISSUE THREAD

**Issue by ipoletaev** [fchollet/keras#4290]
Hi there! During experiments with Keras I have faced with the following problem. When I use any type of RNN layers, method `model.predict(X, batch_size=b_size)` gives different answers with different `b_size` values. Furthermore the bigger the `b_size` value then better the results. It is very strange, because, obviously, if you don't use stateful RNN then output should not depend on the `b_size`.
I wonder if anyone else has stumbled onto that. Thanks for help!

**Comment by keunwoochoi** [comment]
And there's no batch normalization in the network?

**Comment by ipoletaev** [comment]
@keunwoochoi, thanks for comment. Yes, n. network has few batch normalization layers with `mode=2`. So, I'd like to clarify: during the inference batch normalization layers corrects outputs of respective neurons depending on input data, therefore results may be different with different batch size?

**Comment by carlthome** [comment]
Derp. Obviously the results will depend on your batch size with mode=2! If the batch size is too small, the mean and variance for any particular batch will probably not be the same as the full dataset's very often. Increase the batch size or don't use batch normalization in test time.
